core/client/src/main/java/alluxio/client/netty/NettyRemoteBlockReader.java

```diff
@@ -46,41 +44,29 @@
 public final class NettyRemoteBlockReader implements RemoteBlockReader {
   private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
 
-  private final Callable<Bootstrap> mClientBootstrap;
   /** A reference to read response so we can explicitly release the resource after reading. */
   private RPCBlockReadResponse mReadResponse = null;
 
   /**
    * Creates a new {@link NettyRemoteBlockReader}.
    */
-  public NettyRemoteBlockReader() {
-    mClientBootstrap = NettyClient.bootstrapBuilder();
-  }
-
-  /**
-   * Constructor for unittest only.
-   *
-   * @param clientBootstrap bootstrap class of the client channel
-   */
-  public NettyRemoteBlockReader(final Bootstrap clientBootstrap) {
-    mClientBootstrap = new Callable<Bootstrap>() {
-      @Override
-      public Bootstrap call() {
-        return clientBootstrap;
-      }
-    };
-  }
+  public NettyRemoteBlockReader() {}
 
   @Override
   public ByteBuffer readRemoteBlock(InetSocketAddress address, long blockId, long offset,
       long length, long lockId, long sessionId) throws IOException {
-    SingleResponseListener listener = null;
     Channel channel = null;
+    ClientHandler clientHandler = null;
     Metrics.NETTY_BLOCK_READ_OPS.inc();
     try {
-      channel = BlockStoreContext.acquireNettyChannel(address, mClientBootstrap);
-      listener = new SingleResponseListener();
-      channel.pipeline().get(ClientHandler.class).addListener(listener);
+      channel = BlockStoreContext.acquireNettyChannel(address);
```

Comment:
channel is closable itself. I feel it is confusing to allocate another object wrapping on it so that the channel is released when the wrapper object is closed (similar as what you did for FileSystemMasterClient).